packages/flutter/lib/src/cupertino/text_form_field_row.dart

```diff
@@ -0,0 +1,381 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'form_row.dart';
+import 'text_field.dart';
+import 'theme.dart';
+
+/// Creates a [CupertinoFormRow] containing a [FormField] that wraps
+/// a [CupertinoTextField].
+///
+/// A [Form] ancestor is not required. The [Form] simply makes it easier to
+/// save, reset, or validate multiple fields at once. To use without a [Form],
+/// pass a [GlobalKey] to the constructor and use [GlobalKey.currentState] to
+/// save or reset the form field.
+///
+/// When a [controller] is specified, its [TextEditingController.text]
+/// defines the [initialValue]. If this [FormField] is part of a scrolling
+/// container that lazily constructs its children, like a [ListView] or a
+/// [CustomScrollView], then a [controller] should be specified.
+/// The controller's lifetime should be managed by a stateful widget ancestor
+/// of the scrolling container.
+///
+/// The [prefix] parameter is displayed at the start of the row. Standard iOS
+/// guidelines encourage passing a [Text] widget to [prefix] to detail the
+/// nature of the input.
+///
+/// The [margins] parameter is used to pad the contents of the row. It is
+/// directly passed to [CupertinoFormRow]. If the [margins]
+/// parameter is null, [CupertinoFormRow] constructs its own default
+/// margins (which are the standard form row margins in iOS.) If no edge
+/// insets are intended, explicitly pass [EdgeInsets.zero] to [margins].
+///
+/// If a [controller] is not specified, [initialValue] can be used to give
+/// the automatically generated controller an initial value.
+///
+/// Remember to call [TextEditingController.dispose] of the [controller], if one
+/// is specified, when it is no longer needed. This will ensure we discard any
```

Comment:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use-the-passive-voice-recommend-do-not-require